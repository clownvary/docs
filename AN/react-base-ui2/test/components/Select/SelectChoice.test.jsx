import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Choice from 'src/components/Select/SelectChoice';

describe('components/Select/SelectChoice', () => {
  it('component renders fine', () => {
    const props = {
      prefixCls: 'test-choice',
      last: true,
      text: 'mutli-select choice1',
      value: 'msc1'
    };
    const snapshot = renderer.create(<Choice {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('component renders fine with customize choice renderer', () => {
    const props = {
      text: 'mutli-select choice2',
      value: 'msc2',
      choiceRenderer: choiceProps => (<div className="customize-choice">{choiceProps.text} {choiceProps.value}</div>)
    };
    const snapshot = renderer.create(<Choice {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  it('component works fine', () => {
    const onChoiceRemove = jest.fn();
    const props = {
      prefixCls: 'test-choice',
      last: true,
      text: 'mutli-select choice3',
      value: 'msc3',
      onChoiceRemove
    };

    const component = mount(<Choice {...props} />);
    const instance = component.instance();

    instance.onChoiceRemove({ preventDefault: () => {}, stopPropagation: () => {} });
    expect(onChoiceRemove).toHaveBeenCalled();
  });
});
