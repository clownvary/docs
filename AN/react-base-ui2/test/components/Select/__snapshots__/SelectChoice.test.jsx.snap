// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Select/SelectChoice component renders fine 1`] = `
<li
  className="test-choice-selection-choice test-choice-selection-choice__last"
>
  <div
    className="an-tag an-tag__default an-tag__size-m test-choice-selection-choice__tag an-tag__closable"
    closable={true}
    onClose={[Function]}
    prefixCls="an-tag"
    size="m"
    type="default"
  >
    <span
      className="test-choice-selection-choice__content"
    >
      mutli-select choice1
    </span>
    <i
      className="icon icon-close"
      onMouseDown={[Function]}
    />
  </div>
</li>
`;

exports[`components/Select/SelectChoice component renders fine with customize choice renderer 1`] = `
<li
  className="undefined-selection-choice"
>
  <div
    className="an-tag an-tag__default an-tag__size-m undefined-selection-choice__tag an-tag__closable"
    closable={true}
    onClose={[Function]}
    prefixCls="an-tag"
    size="m"
    type="default"
  >
    <div
      className="customize-choice"
    >
      mutli-select choice2
       
      msc2
    </div>
    <i
      className="icon icon-close"
      onMouseDown={[Function]}
    />
  </div>
</li>
`;
